docs/content/docs/libs/state_processor_api.md

```diff
@@ -514,3 +512,120 @@ savepointWriter
+| fields.#.map-key-format | optional | (none)  | String | Defines the format class scheme for decoding map value key data (for ex. java.lang.Long). When It's not provided then it tries to be inferred from the SQL type (only primitive types supported). |
+| fields.#.value-format | optional | (none)  | String | Defines the format class scheme for decoding value data (for ex. java.lang.Long). When It's not provided then it tries to be inferred from the SQL type (only primitive types supported). |
+
+### Default Data Type Mapping
+
+The state SQL connector is inferring the data type for primitive types when `fields.#.value-format` and `fields.#.map-key-format`
+are not defined. The following table shows the `Flink SQL type` -> `Java type` default mapping. If the mapping is not calculated properly
+then it can be overridden with the two mentioned config parameters on a per column bases.
+
+| Flink SQL type          | Java type                                                               |
+|-------------------------|-------------------------------------------------------------------------|
+| CHAR / VARCHAR / STRING | java.lang.String                                                        |
+| BOOLEAN                 | boolean                                                                 |
+| BINARY / VARBINARY      | byte[]                                                                  |
+| DECIMAL                 | org.apache.flink.table.data.DecimalData                                 |
+| TINYINT                 | byte                                                                    |
+| SMALLINT                | short                                                                   |
+| INTEGER                 | int                                                                     |
+| BIGINT                  | long                                                                    |
+| FLOAT                   | float                                                                   |
+| DOUBLE                  | double                                                                  |
+| DATE                    | int                                                                     |
+| INTERVAL_YEAR_MONTH     | long                                                                    |
+| INTERVAL_DAY_TIME       | long                                                                    |
+| ARRAY                   | java.util.List                                                          |
+| MAP                     | java.util.Map                                                           |
+| ROW                     | java.util.List\<org.apache.flink.table.types.logical.RowType.RowField\> |
+
+SHORTCUT: When a complex java class is defined in a column with STRING SQL type then the class instance
+`toString` method result will be the column value. This can be useful when only explanatory query needs to be
```

Comment:
nit: 
This can be useful when only explanatory query needs to be developed in short time. ->
This can be useful when a quick explanatory query is required.